internal/services/legacy/virtual_machine_scale_set_resource_test.go

```diff
@@ -3530,6 +3530,9 @@ resource "azurerm_application_gateway" "test" {
     timeout             = 120
     interval            = 300
     unhealthy_threshold = 8
+    match {
+      status_code = ["200-399"]
+    }
```

Comment:
I think this default value was recently added by Azure Team because other `azurerm_application_gateway.probe` related testcases like `TestAccApplicationGateway_probesWithPort` also failed due to this reason. However, this default value is defined in the [document ](https://learn.microsoft.com/en-us/rest/api/application-gateway/application-gateways/create-or-update?view=rest-application-gateway-2024-03-01&tabs=HTTP#applicationgatewayprobehealthresponsematch)which is by design.